src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -104,6 +105,26 @@ private static void printExitMessage(@Nullable Project project, @Nullable Module
     });
   }
 
+  private static void setPubInProgress(boolean inProgress) {
+    try {
+      DartPubActionBase.class.getMethod("setIsInProgress", boolean.class).invoke(null, inProgress);
+    }
+    catch (Exception e) {
```

Comment:
Perhaps catch `Throwable`? This would catch Errors as well, which ones that would otherwise be compile time errors probably are.